### ISSUE TITLE
Implement more overloads for collections

### ISSUE THREAD

**Issue by MarkusAmshove** [MarkusAmshove/Kluent#108]
The collection assertions should be usable with different approaches, so that a user can decide what is the most readable for him.

I found this while porting the primitive arrays to the common platform module, where I had to write

```
val anArray = intArrayOf(1, 2, 5)
anArray.shouldContainSome(intArrayOf(1, 5))
```

It would be nicer to have different overloads, e.g.:

```
anArray.shouldContainSome(1, 5) // varargs
anArray.shouldContainSome(listFromDb) // Iterable
anArray.shouldContainSome(anotherPrimitiveIntArray) // What we have today
```

**Comment by MarkusAmshove** [comment]
`vararg` won't work for primitive Arrays, because e.g.
```
fun a(args: IntArray)
fun a(vararg args: Int)
```

will have the same JVM signature

**Comment by MarkusAmshove** [comment]
This is within the just released 1.39
